### PR TITLE
[Frontend] Add sampling parameters to Responses API

### PR DESCRIPTION
**Pull request by DanielMe**
## Purpose

Port essential sampling parameters from `/v1/chat/completions` to `/v1/responses` API to provide basic generation control for users of the Responses API.

**Added Parameters:**
- ~~`presence_penalty`, `frequency_penalty`,~~ `repetition_penalty` - repetition control
- ~~`min_p`,~~ `seed`, `stop`, `ignore_eos`, ~~`min_tokens`~~ - sampling control
- ~~`prompt_logprobs`, `spaces_between_special_tokens`~~ - ~~output control~~
- ~~`include_stop_str_in_output`, `truncate_prompt_tokens`~~ - ~~formatting control~~
- ~~`logits_processors`, `allowed_token_ids`,~~ `vllm_xargs` - advanced control

**Key Changes:**
- Extended `ResponsesRequest` protocol with 5 core sampling parameters
- ~~Added `logits_processor_pattern` parameter to `to_sampling_params()` method~~

## Test Plan

**Unit Tests:**

Adding unit tests just to validate the API mapping:

```bash
pytest tests/entrypoints/openai/responses/test_sampling_params.py -v
```

**Integration Test:**

Adding an integration test to prove that the end-to-end call works with the new parameters.
```bash
pytest tests/entrypoints/openai/responses/test_simple.py::test_extra_sampling_params -v
```

## Test Result

```
tests/entrypoints/openai/responses/test_sampling_params.py::TestResponsesRequestSamplingParams::test_basic_sampling_params PASSED                   [ 20%]
tests/entrypoints/openai/responses/test_sampling_params.py::TestResponsesRequestSamplingParams::test_extra_sampling_params PASSED                   [ 40%]
tests/entrypoints/openai/responses/test_sampling_params.py::TestResponsesRequestSamplingParams::test_stop_string_conversion PASSED                  [ 60%]
tests/entrypoints/openai/responses/test_sampling_params.py::TestResponsesRequestSamplingParams::test_default_values PASSED                          [ 80%]
tests/entrypoints/openai/responses/test_sampling_params.py::TestResponsesRequestSamplingParams::test_seed_bounds_validation PASSED                  [100%] 
```

```
tests/entrypoints/openai/responses/test_simple.py::test_basic[Qwen/Qwen3-8B] PASSED                                                                                                 [ 16%]
tests/entrypoints/openai/responses/test_simple.py::test_enable_response_messages[Qwen/Qwen3-8B] PASSED                                                                              [ 33%]
tests/entrypoints/openai/responses/test_simple.py::test_reasoning_item[Qwen/Qwen3-8B] PASSED                                                                                        [ 50%]
tests/entrypoints/openai/responses/test_simple.py::test_streaming_output_consistency[Qwen/Qwen3-8B] PASSED                                                                          [ 66%]
tests/entrypoints/openai/responses/test_simple.py::test_max_tokens[Qwen/Qwen3-8B] PASSED                                                                                            [ 83%]
tests/entrypoints/openai/responses/test_simple.py::test_extra_sampling_params[Qwen/Qwen3-8B] PASSED                                                                                 [100%]
```

---
<details>
<summary> Essential Elements of an Effective PR Description Checklist </summary>

- [x] The purpose of the PR, such as "Fix some issue (link existing issues this PR will resolve)".
- [x] The test plan, such as providing test command.
- [x] The test results, such as pasting the results comparison before and after, or e2e results
- [ ] (Optional) The necessary documentation update, such as updating `supported_models.md` and `examples` for a new model.
- [ ] (Optional) Release notes update. If your change is user facing, please update the release notes draft in the [Google Doc](https://docs.google.com/document/d/1YyVqrgX4gHTtrstbq8oWUImOyPCKSGnJ7xtTpmXzlRs/edit?tab=t.0).
</details>

### CODE REVIEW

File: vllm/entrypoints/openai/responses/protocol.py [hunk 1]
```diff
@@ -220,6 +225,39 @@ class ResponsesRequest(OpenAIBaseModel):
     # this cannot be used in conjunction with previous_response_id
     # TODO: consider supporting non harmony messages as well
     previous_input_messages: list[OpenAIHarmonyMessage | dict] | None = None
+
+    presence_penalty: float | None = 0.0
+    frequency_penalty: float | None = 0.0
```
[the remark this answers is not in the document]
Comment (DanielMe) [reply vllm/entrypoints/openai/responses/protocol.py]: The reason for the original implementation is that it is consistent with how it is handled in `/chat/completions`:
https://github.com/DanielMe/vllm/blob/4bd4bc43b53e003551b9908ced932e642270cbea/vllm/entrypoints/openai/chat_completion/protocol.py#L162

However I agree with the AI feedback that it is better to keep things consistent and to allow initialization via `default_sampling_params` like with other parameters

**Comment by DanielMe** [comment]
Thanks @qandrew ! This PR comes out of a concrete need for `stop` and `vllm_xargs` primarily and then, a bit less important for our use case, `seed`, `repetition_penalty`, `prompt_logprobs` and `ignore_eos` for running a custom model. We have a preference for using the Responses API over Chat Completions because of the better structured output and the momentum behind this API.

While working on this proposal I figured that other people may find use for the other sampling params already supported by Chat Completions and they turned out easy to port.

That being said, I can see the case for going on a case-by-case basis depending on concrete need to avoid committing to future maintenance of an attribute nobody asked for.

I could reduce the PR to only the six fields for which we have concrete use cases or only the two fields which are the most important ones?

File: vllm/entrypoints/openai/responses/protocol.py [hunk 2]
```diff
@@ -228,6 +231,18 @@ class ResponsesRequest(OpenAIBaseModel):
     # this cannot be used in conjunction with previous_response_id
     # TODO: consider supporting non harmony messages as well
     previous_input_messages: list[OpenAIHarmonyMessage | dict] | None = None
+
+    repetition_penalty: float | None = None
```
Comment (chaunceyjiang) [inline vllm/entrypoints/openai/responses/protocol.py:237]: Compared to the previous version, these additions make it much more reasonable.  


Let's invite others to review this. 

/cc @qandrew @yeqcharlotte @DarkLight1337

**Review by qandrew — APPROVED** [review]
lgtm, thanks!

**Comment by chaunceyjiang** [comment]
@qandrew ok

**Review by DarkLight1337 — APPROVED** [review]
